### ISSUE TITLE
Feature request: Add Wilder's Smoothing / Moving Average (rma)

### ISSUE THREAD

**Issue by jp-0704** [pola-rs/polars#29164]
### Description

Polars currently provides `ewm_mean()`, but implementing technical indicators that use Wilder's smoothing requires additional manual handling because Wilder's method differs from a conventional EWM primarily in its initialization.

Wilder's smoothing with period `n` is equivalent to an EMA with:

$$
\alpha = \frac{1}{n}
$$

and the recursive update:

$$
S_t = \frac{(n-1)S_{t-1}+X_t}{n}
$$

However, the initial value is defined as the simple arithmetic mean of the first `n` observations:

$$
S_n = \frac{1}{n}\sum_{i=1}^{n}X_i
$$

After this initial value, the recursive Wilder formula is applied.

## Problem

Using:

```python
series.ewm_mean(
    alpha=1 / n,
    adjust=False,
    min_samples=n
)
```

provides the correct recursive smoothing factor, but does not provide a direct way to specify the Wilder initialization.

`min_samples=n` only controls when the EWM starts returning non-null values; it does not seed the recursive calculation with the SMA of the first `n` observations.

This becomes a recurring issue when implementing standard technical indicators.

For example, ATR requires:

```text
True Range
    ↓
first ATR = SMA(first n True Range values)
    ↓
Wilder recursive smoothing
```

Similarly, RSI requires Wilder smoothing independently on gains and losses:

```text
price changes
    ├── gains → first average gain = SMA(first n gains)
    │                    ↓
    │              Wilder smoothing
    │
    └── losses → first average loss = SMA(first n losses)
                         ↓
                   Wilder smoothing
```

Both indicators therefore require essentially the same underlying operation.

## Proposed API

A reusable Wilder moving average / RMA operation could be exposed, for example:

```python
series.rma(period=n)
```

or:

```python
series.ewm_mean(
    alpha=1 / n,
    adjust=False,
    min_samples=n,
    initialization="sma"
)
```

The exact API is open to discussion.

The important requirement is that the operation should:

1. Calculate the SMA of the first `n` valid observations.
2. Use that SMA as the initial smoothed value.
3. Apply:

$$
S_t = \frac{(n-1)S_{t-1}+X_t}{n}
$$

   for subsequent observations.
4. Preserve appropriate null behavior when there are insufficient observations.

## Why a general-purpose operation would be useful

A first-class Wilder/RMA operation could be reused across many financial indicators, including:

- ATR
- RSI
- ADX
- +DI / -DI
- Williams-style smoothing and related indicators
- Other indicators based on Wilder's moving average

Instead of each implementation manually reproducing the initialization and recursive calculation, the same well-tested Polars primitive could be used consistently.

For example, RSI could conceptually become:

```python
avg_gain = gain.rma(14)
avg_loss = loss.rma(14)
```

and ATR:

```python
atr = true_range.rma(14)
```

This would also avoid relying on implementation-specific workarounds involving `shift`, `fill_null`, or Python-side loops for a fundamentally standard rolling/smoothing operation.

## Expected benefit

A first-class Wilder smoothing operation would make Polars more suitable for vectorized financial/technical-analysis workloads while also providing a single, reusable implementation of Wilder's initialization and recursive smoothing semantics.
